src/pybind/mgr/dashboard_v2/frontend/src/app/ceph/cluster/osd/osd-list/osd-list.component.ts

```diff
@@ -53,4 +57,9 @@ export class OsdListComponent {
       });
     });
   }
+
+  collectStates(osd) {
+    const states = ['up', 'in', 'down', 'out'];
+    return Object.keys(osd).filter(key => states.indexOf(key) !== -1);
```

Comment:
This method will always return `up` and `in`.
If an OSD is down it will return `up: 0`, but this method will see that there is an `up` property and display the `up` state.

I would suggest to just check the values of `up` and `in` and determine the states with that.